APMrover2/Log.cpp

```diff
@@ -510,9 +510,9 @@ void Rover::Log_Write_Vehicle_Startup_Messages()
 // start a new log
 void Rover::start_logging()
 {
-    in_mavlink_delay = true;
+    DataFlash.EnableWrites(false);
     DataFlash.StartUnstartedLogging();
-    in_mavlink_delay = false;
+    DataFlash.EnableWrites(true);
```

Comment:
... also, while all of this is a little clean-uppy... Consider that currently library functions indirectly call WritePrioritisedBlock - meaning they're no doing the whole "am I doing the mavlink delay thing".  I'm not exactly sure how terribly bad that is, but with these changes they can start doing `DataFlash.should_log()`...